### ISSUE TITLE
Error with Stripe payment

### ISSUE THREAD

**Issue by Mullycap** [mirumee/saleor#392]
Hi:
I'm using the Stripe for payment but I found a problem, when I pay the order with Stripe, in test mode, I finish the process but in the order view in status it says waiting for payment, but when I go to the dashboard in the payment status says completed.
When I do the process again the status change for fully paid, but it charges two times to the costumer.
The payment in my setting is the following:

PAYMENT_BASE_URL = 'http://%s/' % CANONICAL_HOSTNAME

PAYMENT_MODEL = 'order.Payment'
PAYMENT_HOST = 'localhost:8000'

The variants work properly.
Best regards

**Comment by artursmet** [comment]
Which Payment provider are you using, there are two `StripeProvider` and `StripeCardProvider`
Also could you check payment status in stripe's dashboard?
When did you cloned saleor repository?

**Comment by Mullycap** [comment]
Im using StripeProvider I will change to StripeCardProvider to se if works better

**Comment by Mullycap** [comment]
If I use StripeCardProvider the button to pay doesn't appear, only the form to fill de info, could be that I don't have any SSL certificate?

**Comment by patrys** [comment]
Do you have form.media included in the template? Are there any errors in the browser's console?

**Comment by artursmet** [comment]
StripeCardProvider reders only form with credit card details (owner name, card number, cvc, expire date). Any payment provider doesn't require SSL certificate.
I tested stripe payments on fresh saleor installation and it looks ok. That's I asked you to check if transaction was processed properly in Stripe (in their dashboard).

**Comment by Mullycap** [comment]
With the previous one the payment appear correctly in the dashboard of stripe. The only problem was that in the dashboard of saleor it appear with the payment completed but with the status of the order in waiting for payment.
Related with the template didn't change any of the templates and in the stripe.html it appear the form.media.
Now when I'm trying to do the payment appear like this, without a button and in the browser's console I don't have any problem, I use chrome for development:
![screen shot 2015-11-18 at 22 52 32](https://cloud.githubusercontent.com/assets/7394944/11255540/1ed560da-8e47-11e5-88c5-4e7e3e72e7be.png)
With the data fulfilled the button doesn't appear. Like I said before I didn't change any of the code of the stripe.html template in order/payments
Seriously thanks a lot for your help.

**Comment by Mullycap** [comment]
I don't know if it helps, in settings the Stripe configuration is the following:
tripe': ('payments.stripe.StripeCardProvider', {
                'secret_key': 'SK_TEST_KEY',
                'public_key': 'PK_TEST_KEY'})

**Comment by artursmet** [comment]
Let me describe how status flow is processing, maybe we will fix it this way.
* After successful checkout, order is placed with status `payment-pending` https://github.com/mirumee/saleor/blob/master/saleor/order/views.py#L85
* Then user is able to select payment method (Stripe in your scenario)
* Stripe button opens a modal window, with credit card details form
* After successful form submission payment form is saved (https://github.com/mirumee/django-payments/blob/master/payments/stripe/__init__.py#L30) in save method money is captured (by default) https://github.com/mirumee/django-payments/blob/master/payments/stripe/forms.py#L64 and when amount is greater than zero, payment status is beeing changed: https://github.com/mirumee/django-payments/blob/master/payments/models.py#L154
* Status change emits a signal (https://github.com/mirumee/django-payments/blob/master/payments/models.py#L105) and this signal is received in Order utils (https://github.com/mirumee/saleor/blob/master/saleor/order/utils.py#L32)
* Finally order should have status `fully-paid`

Please follow these steps and check if you have everything in place in your code (signal related part is most important). 
I spawned fresh installation on heroku right now and I'm unable to reproduce this issue. So I think that you can have some bug in your code, or older version somewhere (with this bug).

Test instance is here: https://saleordebug.herokuapp.com

In payment method step there are two choices available (second one doesn't have proper label due to css bug). First is stripe with modal, and second one is `StripeCardProvider`

**Comment by patrys** [comment]
@Mullycap, were you able to pinpoint the problem? Can we assist you somehow?

**Comment by Mullycap** [comment]
I have been reviewing all the code and I updated to the new version, in the one I used the methods where in the init.py and I created a utils.py like in the version that you had in the github nowadays, but the problem persisted, but I detect it.
The problem seems that was in the method that calculate the delivery, we had a little bug that with a weight that wasn't in the table of prices that the delivery company gave us the formula returns a negative value for the delivery, for example product 5€ + delivery -1 = total 4€, with that negative value in the delivery we don't know why in the dashboard appeared like waiting for payment but the stripe capture the payment.
When we solve that problem getting the correct cost of the delivery now the formula again give the correct amount and work perfectly.
Thanks for everything, seriously I'm surprised with this awesome support.
